registry/registry.go

```diff
@@ -150,3 +154,48 @@ func (z *zookeeperRegistry) deleteNode(path string) error {
 	}
 	return nil
 }
+
+func (z *zookeeperRegistry) keepalive(path string, content []byte) {
+	sessionID := z.conn.SessionID()
+	ticker := time.NewTicker(z.sessionTimeout / 2)
+	defer ticker.Stop()
+	for range ticker.C {
+		cur := z.conn.SessionID()
+		// sessionID changed
+		if cur > 0 && sessionID != cur {
+			// re-ensureName
+			if err := z.ensureName(path, content, zk.FlagEphemeral); err != nil {
+				return
```

Comment:
If we return, does the `keepalive` still work? If the zk server shutdown, it will return err.

Reply:
I check th zk lib. We should disable keepalive when the connection is closed. We should alway try to keep the server registered when the servers shutdown.

Reply:
failed keepalive when ErrConnectionClosed && only keepalive when Register success

Reply:
If the zkServer shutdown and restart, the keepalive won't work anymore.

Reply:
We should reconstruct the connection when the zk.ErrConnectionClosed happens?

Reply:
I mean that the zkServer is down, not our registry client. 
My thought: we maintain the service info local, and listen to the connection change event, when received the connection change event, to check if the service is still registered, if not registered, re-register.
But now, the zk lib didn't support listening to connection change events. Using a timer to check the session change or not is also a good way. We needn't cancel the timer task, there is one timer task to check the connection is changed is enough.

Reply:
ok. Please review the final change.